ls1intum/Artemis
# `General`: Ensure manually created users can log in

### Checklist
#### General


#### Server

### Motivation and Context
When manually creating users as admin, the admin can select from a few roles the user can have. By default none of the options is selected. However, to be able to construct a valid token for login the user needs to have at lease one role. This is typically the STUDENT/USER role that has no special permissions.

### Description
Updated the user creation logic to always include at least this role if no other ones have been defined for the user.

### Steps for Testing

Prerequisites:
- 1 Admin

1. As admin, go to the user management page.
2. Create a new user. You can select a set of roles for the new user.
3. Select none of the roles, the newly created user should still be able to log in to Artemis afterwards.

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| UserCreationService.java | 94% | ✅                           |
